### ISSUE TITLE
Keyboard navigation in Modal doesn't work as expected

### ISSUE THREAD

**Issue by orionmiz** [formaat-design/reshaped#209]
**Describe the bug**
Keyboard navigation (especially using `Tab`) between radio inputs within the modal is different from the behavior in a location outside the modal.

**To Reproduce**
[Codesandbox](https://codesandbox.io/p/sandbox/reshaped-keyboard-navigation-in-modal-issue-lf7klg?file=%2Fsrc%2FApp.tsx%3A1%2C1-78%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clr7ecwnk00063b6fk7e87a1t%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clr7ecwnj00023b6f9hg0rh0i%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clr7ecwnj00033b6fewotarfn%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clr7ecwnj00053b6f7j6s2mi3%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clr7ecwnj00023b6f9hg0rh0i%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clr7ecwnj00013b6fnimp3m7g%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpublic%252Findex.html%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clr7qb26o00je3b6fgg83ilt5%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A1%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A78%252C%2522endColumn%2522%253A1%257D%255D%257D%255D%252C%2522id%2522%253A%2522clr7ecwnj00023b6f9hg0rh0i%2522%252C%2522activeTabId%2522%253A%2522clr7qb26o00je3b6fgg83ilt5%2522%257D%252C%2522clr7ecwnj00053b6f7j6s2mi3%2522%253A%257B%2522id%2522%253A%2522clr7ecwnj00053b6f7j6s2mi3%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clr7ecwnj00043b6fw3e2curl%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clr7ecwnj00043b6fw3e2curl%2522%257D%252C%2522clr7ecwnj00033b6fewotarfn%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clr7ecwnj00033b6fewotarfn%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A20.736137667304007%257D)

**Expected behavior**
If you press the `Tab` while the radio input is focused, the focus should move to the button

**Screenshots**

https://github.com/formaat-design/reshaped/assets/39466936/5e256bb7-58b5-4782-9dcd-5a19b9691f46

**Environment (please complete the following information):**
 - OS: MacOS
 - Browser: Chrome 120.0.6099.216
 - Version: latest

**Comment by blvdmitry** [comment]
https://github.com/formaat-design/reshaped/assets/887379/fcdca75d-d5c4-4092-bce6-977845304425

I have just released 2.8.2 that supports the behavior on the video. When I navigate the input selection there is happens with arrow keys. 

In case you are updating from 2.8.0 or below, it also has some changes added in 2.8.1 regarding using the `exports` filed in `package.json`. It's tested with our starter examples but just please let us know if it's not working smoothly for you.
